polaris-react/src/components/DataTable/DataTable.scss

```diff
@@ -24,6 +24,10 @@ $breakpoint: 768px;
 
 .Navigation {
   display: none;
+
+  &.IncreasedTableDensity {
+    padding: var(--p-space-2) var(--p-space-4) var(--p-space-1) var(--p-space-2);
```

Comment:
Looking at the figma, I think we want it to be 3 (12px) 2 (8px) 1 (4px) and then the last one which can probably be 2 (8px). 

https://screenshot.click/13-45-gl6vd-lsrwy.png

Let me know if I'm looking at this wrong.

Reply:
@mirualves 's verdict for the top: `--p-space-2`